feat: policy page

k3nt0w opened the pull request:
## Proposed Changes

- [x] Implement policy page component
  - policy list
  - link button on header
- [x] Connect component with datasource
- [x] test code

## Implementation

### Policy List Page


![スクリーンショット 2020-07-13 5 25 48](https://user-images.githubusercontent.com/16770233/87255865-6a5a6100-c4c9-11ea-9467-e09fad38ca96.png)

![スクリーンショット 2020-07-13 5 25 39](https://user-images.githubusercontent.com/16770233/87255867-6cbcbb00-c4c9-11ea-8f94-492b4b4da625.png)

## ❓ Question 

@aggre 
I need to do the "createGetVotablePolicy" I couldn't figure out how to get a "PolicySetContract". I am trying to implement a function to solve this problem by referring to the dev-kit repositories. However, in the end I couldn't solve the problem. Please advise me as I am commenting on the relevant section. If you don't understand the error in the code I tried, please uncomment out and run it:pray: After the correct implementation is completed, I will implement the test.

k3nt0w commented on packages/web/src/fixtures/dev-kit/client.ts:
```diff
@@ -163,3 +175,18 @@ export const holdersShare = async (amount: string, lockedups: string) => {
+  if (web3Client) {
+    // FIXME(@k3nt0w)
+    // return devKitClient.createGetVotablePolicy(getPolicySetContract(web3Client))()
+    return [
+      '0xCbC478710D3aCD2E7B94F58BB639f345af5f7869',
+      '0xCbC478710D3aCD2E7B94F58BB639f345af5f7869',
+      '0xCbC478710D3aCD2E7B94F58BB639f345af5f7869',
+      '0xCbC478710D3aCD2E7B94F58BB639f345af5f7869'
+    ]
```
To check the operation of the Component, a fixed policy address is returned here.

k3nt0w commented on packages/web/src/fixtures/dev-kit/get-policy-set-contract.ts:
```diff
@@ -0,0 +1,8 @@
+// eslint-disable-next-line
+export const getPolicySetContract = (client: Web3, address?: string, options?: any) => {}
+// FIXME(@k3nt0w): this is test code for confirmation operation.
+// createPolicySetContract(client)(address, options)
```
I couldn't get `PolicySetContract`... please give me some advice:bow:

aggre replied:
Thank you for the implementation!

To generate a PolicySet contract instance, you can use the familiar codes in "fixtures/dev-kit/client.ts" to create, and it is.

Like: https://github.com/dev-protocol/stakes.social/blob/270551f96abb332f37f47bf688bc59dc8d66f69c/packages/web/src/fixtures/dev-kit/client.ts#L16

So, for example:
```ts
client.policySet(await getContractAddress(client, 'policySet'))
```

Can generate that like this. :)

k3nt0w replied:
I was completely wrong in my perception.... I seem to have been rather confused by what I read in dev-kit Interface:cry: Thank you for your advice with specific code!! I'll fix it soon.

aggre replied:
I'm sorry for the lack of documentation 🙇‍♂️ Many thanks!

k3nt0w commented:
@aggre 

I am very sorry for the delay in reflecting the review:bow:

I have followed your advice and modified the implementation. 
https://github.com/dev-protocol/stakes.social/blob/471ee3a4b1990f4942d19b1f5c4599bf7648ae5e/packages/web/src/fixtures/dev-kit/client.ts#L170
But, this `policySet` function doesn't have `createGetVotablePolicy`. From reading the code, it looks like `DevkitContract` doesn't have a function for `createGetVotablePolicy`. My confusion still persists, it seems...

I've added test code and refactored it, so please advise me about this part again, I'm sorry :pray:

k3nt0w commented on packages/web/src/fixtures/github/cache-path.ts:
```diff
@@ -1,4 +1,6 @@
-    `https://raw.githubusercontent.com/dev-protocol/assets/main/market/${marketAddress}/info.json`
+    `https://raw.githubusercontent.com/dev-protocol/assets/main/market/${marketAddress}/info.json`,
+  getPolicyInformation: (policyAddress: string) =>
+    `https://raw.githubusercontent.com/dev-protocol/assets/main/market/${policyAddress}/info.json`
```
I replaced target branch from master to main.

aggre replied:
Thank you!

aggre commented:
@k3nt0w I fixed the calling method for getting policy addresses by https://github.com/dev-protocol/stakes.social/pull/368/commits/c9ddff80c17df725bf5674e1c0ac0e79a3d82659.

I'm sorry for specs are hard to convey!

k3nt0w commented:
@aggre 
Thank you very much🙇‍♂️ I'll fix CI now！

k3nt0w commented:
@aggre 
I fixed test suite just now! Please check the operation:bow:

aggre commented:
@k3nt0w Many thanks! I fixed a bit, so I merge this when after passed the CI.
